src/main/java/es/tid/pce/pcep/objects/tlvs/SRPolicyCandidatePathIdentifiersTLV.java

```diff
@@ -0,0 +1,196 @@
+   specified in [I-D.ietf-spring-segment-routing-policy] Section 2.4.
+
+   Discriminator: 32-bit value that encodes the Discriminator of the
+   candidate path.
+   
+	 */
+	
+	private int protocol;
+	
+	private int reserved;
+	
+	private long originatorASN;
+	
+	private Inet4Address originatorAddress;
+	
+	private long discriminator;
+	
+	public SRPolicyCandidatePathIdentifiersTLV(){
+		this.TLVType=57;
+	}
+	
+	public SRPolicyCandidatePathIdentifiersTLV(byte[] bytes, int offset)throws MalformedPCEPObjectException{		
+		super(bytes,offset);		
+		decode();
+	}
+	
+	public int getProtocol() {
+		return protocol;
+	}
+
+	public void setProtocol(int protocol) {
+		this.protocol = protocol;
+	}
+	
+	
+
+	public int getReserved() {
+		return reserved;
+	}
+
+	public void setReserved(int reserved) {
+		this.reserved = reserved;
+	}
+
+	public long getOriginatorASN() {
+		return originatorASN;
+	}
+
+	public void setOriginatorASN(long originatorASN) {
+		this.originatorASN = originatorASN;
+	}
+
+	public Inet4Address getOriginatorAddress() {
+		return originatorAddress;
+	}
+
+	public void setOriginatorAddress(Inet4Address originatorAddress) {
+		this.originatorAddress = originatorAddress;
+	}
+
+	public long getDiscriminator() {
+		return discriminator;
+	}
+
+	public void setDiscriminator(long discriminator) {
+		this.discriminator = discriminator;
+	}
+	
+	@Override
+	public int hashCode() {
+		final int prime = 31;
+		int result = super.hashCode();
+		result = prime * result + Objects.hash(discriminator, originatorASN, originatorAddress, protocol, reserved);
+		return result;
+	}
+
+	@Override
+	public boolean equals(Object obj) {
+		if (this == obj)
+			return true;
+		if (!super.equals(obj))
+			return false;
+		if (getClass() != obj.getClass())
+			return false;
+		SRPolicyCandidatePathIdentifiersTLV other = (SRPolicyCandidatePathIdentifiersTLV) obj;
+		return discriminator == other.discriminator && originatorASN == other.originatorASN
+				&& Objects.equals(originatorAddress, other.originatorAddress) && protocol == other.protocol
+				&& reserved == other.reserved;
+	}
+	
+
+	@Override
+	public String toString() {
+		return "SRPolicyCandidatePathIdentifiersTLV [protocol=" + protocol + ", reserved=" + reserved
+				+ ", originatorASN=" + originatorASN + ", originatorAddress=" + originatorAddress + ", discriminator="
+				+ discriminator + "]";
+	}
+
+	@Override
+	public void encode() {
+		//Log.debug("Encoding SRPolicyCandidatePathIdentifiers TLV");
+		this.setTLVValueLength(16);
+		this.tlv_bytes=new byte[this.getTotalTLVLength()];
+		this.encodeHeader();
+		int offset=4;
+		this.reserved=0;
+		
+		ByteHandler.encode1byteInteger(protocol, this.tlv_bytes, offset);
+		offset+=1;
+		ByteHandler.encode3bytesInteger(reserved, this.tlv_bytes, offset);
+		offset+=3;
+		ByteHandler.encode4bytesLong(originatorASN, this.tlv_bytes, offset);
+		offset+=4;
+		System.arraycopy(originatorAddress.getAddress(),0,this.tlv_bytes,offset,4);
```

Comment:
Be careful,  Originator Address: Represented as 128 bit value where IPv4 address
   are encoded in lowest 32 bits, part of the originator identifier, as
   specified in [I-D.ietf-spring-segment-routing-policy] Section 2.4.
You are only econding the 32 bits of the IPv4 address....
 in [I-D.ietf-spring-segment-routing-policy] Section 2.4.:
 Node Address : represented as a 128-bit value.  IPv4 addresses
      MUST be encoded in the lowest 32 bits, and the high-order bits
      MUST be set to zero.
So... set all High order bits to 0 (that is upper 96 bits, about 10 bytes of Zeros)